nixos/modules/services/misc/asusd.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.asusd;
+  tomlFormat = pkgs.formats.toml {};
+in
+{
+  options = {
+    services.asusd = {
+      enable = mkEnableOption ''
+        Provides an interface for rootless control of some system functions most users wish to control such as fan speeds, keyboard LEDs, graphics modes.
+      '';
+      ledmodes = mkOption {
+        type = tomlFormat.type;
+        description = ''
+          <filename>asusd-ledmodes.toml</filename> as a Nix attribute set.
+          See <link xlink:href="https://gitlab.com/asus-linux/asusctl/-/blob/main/data/asusd-ledmodes.toml">asusd-ledmodes.toml</link> for examples.
+        '';
+        default = {};
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Set the ledmodes to the packaged ledmodes by default.
+    services.asusd.ledmodes = mkDefault (
+      let
+        # Convert packaged asusd-ledmodes.toml to JSON.
+        json = pkgs.runCommand "asusd-ledmodes.json" {
+            nativeBuildInputs = [ pkgs.remarshal ];
+          } ''
+            toml2json "${pkgs.asusctl}/etc/asusd/asusd-ledmodes.toml" "$out"
+          '';
+        # Convert JSON to Nix attribute-set.
+        attrs = builtins.fromJSON (builtins.readFile json);
+      in attrs
+    );
+
+    environment.systemPackages = with pkgs; [ asusctl ];
+    services.dbus.packages = with pkgs; [ asusctl ];
+
+    # Avoiding adding udev rules file of asusctl as-is due to the reference to systemd.
+    # services.udev.packages = with pkgs; [ asusctl ];
+
+    # See ${pkgs.asusctl}/lib/udev/rules.d/99-asusd.rules
+    services.udev.extraRules = ''
+      ACTION=="add|change", SUBSYSTEM=="input", ENV{ID_VENDOR_ID}=="0b05", ENV{ID_MODEL_ID}=="1[89][a-zA-Z0-9][a-zA-Z0-9]|193b", ENV{ID_TYPE}=="hid", TAG+="systemd", ENV{SYSTEMD_WANTS}="asusd.service"
+      ACTION=="add|remove", SUBSYSTEM=="input", ENV{ID_VENDOR_ID}=="0b05", ENV{ID_MODEL_ID}=="1[89][a-zA-Z0-9][a-zA-Z0-9]|193b", RUN+="${pkgs.systemd}/bin/systemctl restart asusd.service"
+    '';
+
+    environment.etc."asusd/asusd-ledmodes.toml".source = tomlFormat.generate "asusd-ledmodes.toml" cfg.ledmodes;
+
+    systemd.services.asusd = {
+      description = "Asus Control Daemon";
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "dbus.socket" ];
+      environment.IS_SERVICE = "1";
+      preStart = ''
```

Comment:
Not important and definitely preference... but how do you feel about `ConfigurationDirectory = "asusd";` instead?

Reply:
Good suggestion! I wasn't aware of `ConfigurationDirectory`. Seems better using that, even though `asusd` does not use the `CONFIGURATION_DIRECTORY` env var.
Thanks :+1: